maintain/going-production-with-k8s.md

```diff
@@ -0,0 +1,468 @@
+  Monitor for for Volume Space Low alerts
+  
+  ```
+  30|VolumeSpaceLow|ALARM|VOLUME|Triggered when the free space available in a volume goes below a threshold.
+  ```
+
+* For applications needing node level availability and read parallelism across nodes, it is recommended to set the 
+  volumes with replication factor 2 or replication factor 3
+  
+  Following storeclass shows how to create a SC for a PVC with replication factor of 3.
+ ```yaml
+ kind: StorageClass
+ apiVersion: storage.k8s.io/v1beta1
+ metadata:
+   name: portworx-sc
+ provisioner: kubernetes.io/portworx-volume
+ parameters:
+   repl: "3"
+ ```
+    
+* PX makes best effort to distribute volumes evenly across all nodes and based on the `iopriority` that is requested. When
+  PX cannot find the appropriate media type that is requested to create a given `iopriority` type, it will attempt to
+  create the volume with the next available `iopriority` level. 
+  
+ ```yaml
+ kind: StorageClass
+ apiVersion: storage.k8s.io/v1beta1
+ metadata:
+   name: portworx-sc
+ provisioner: kubernetes.io/portworx-volume
+ parameters:
+   repl: "3"
+   priority_io: "high"
+ ```
+  
+* Volumes can be created in different availability zones by using the `--zones` option in the `pxctl volume create` command
+
+For e.g., for PX cluster running on AWS, 
+
+ ```yaml
+ kind: StorageClass
+ apiVersion: storage.k8s.io/v1beta1
+ metadata:
+   name: portworx-sc
+ provisioner: kubernetes.io/portworx-volume
+ parameters:
+   repl: "3"
+   priority_io: "high"
+   zones: "us-east-1a"
+ ```
+* Volumes can be created in different racks using `--racks` option and passing the rack labels when creating the volume
+
+ ```yaml
+ kind: StorageClass
+ apiVersion: storage.k8s.io/v1beta1
+ metadata:
+   name: portworx-sc
+ provisioner: kubernetes.io/portworx-volume
+ parameters:
+   repl: "3"
+   priority_io: "high"
+   racks: "rack1"
+ ```
+
+* If the volumes need to be protected against accidental deletes because of background garbage collecting scripts, 
+  then the volumes need to enabled with `--sticky` flag
+  
+ ```yaml
+ kind: StorageClass
+ apiVersion: storage.k8s.io/v1beta1
+ metadata:
+   name: portworx-sc
+ provisioner: kubernetes.io/portworx-volume
+ parameters:
+   repl: "3"
+   priority_io: "high"
+   racks: "rack1"
+   sticky: "true"
+ ```
+  
+* For applications that require shared access from multiple containers running in different hosts, 
+  Portworx recommends running  shared volumes. Shared volumes can be configured as follows by adding `shared: "true" ` to the storage class:
+ 
+ ```yaml
+ kind: StorageClass
+ apiVersion: storage.k8s.io/v1beta1
+ metadata:
+   name: portworx-sc
+ provisioner: kubernetes.io/portworx-volume
+ parameters:
+   repl: "3"
+   priority_io: "high"
+   racks: "rack1"
+   shared: "true"
+ ```
+
+  This [page](https://docs.portworx.com/manage/volumes.html) gives more details on different volume types, 
```

Comment:
The page https://docs.portworx.com/manage/volumes.html is not relevant to k8s. It describes pxctl volume create in detail which should not be used in k8s.